packages/components/src/components/as-histogram-widget/utils/draw.service.ts

```diff
@@ -293,9 +296,31 @@ function _delayFn(_d, i) {
   return i;
 }
 
-export function conditionalFormatter(value) {
-  if (value > 0 && value < 1) {
-    return decimalFormatter(value);
+function getFloatPrecision(value) {
+  const expValue = value.toPrecision()
+  const expPos = expValue.indexOf('.')
+  return expPos > -1 ? expValue.slice(expPos + 1).length : 0
+}
+
+function getDomainPrecision(domain) {
```

Comment:
It looks like this logic deserves a bit of testing. Why don't we create some basic specs on formatting, exemplifying several combinations of values & domains?

Reply:
@neokore is this already done?

Reply:
Yes, using the e2e tests I showed to you and @VictorVelarde agrees (https://github.com/CartoDB/airship/pull/687#discussion_r413124631)